lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -749,15 +751,30 @@ data ApiNetworkParameters = ApiNetworkParameters
     , decentralizationLevel :: !(Quantity "percent" Percentage)
     , desiredPoolNumber :: !Word16
     , minimumUtxoValue :: !(Quantity "lovelace" Natural)
-    , hardforkAt :: !(Maybe ApiEpochInfo)
+    , eras :: !ApiEraInfo
```

Comment:
why not to have 
```
eras: [ApiEraInfo]
...
data Eras = Byron | Shelley | Allegra | Mary deriving (Eq, Ord, Show)

data ApiEraInfo = ApiEraInfo {
   hardForkToEra :: !Era
   hardFromEra :: !(Maybe Era)
   timeOfTransition:: !(Maybe ApiEpochInfo)
}
```
?

Reply:
I think it is more robust in a sense that when new hard forks arrive they will just add new element to list. But maybe there is some rationale behind this way of storing era transitions?

Reply:
@KtorZ advocated for an object keyed by era-name since Daedalus wants to do stuff like `isAllegra` and `isShelley`, and already have to be highly aware of e.g. the ordering of them.

Although with https://github.com/input-output-hk/cardano-wallet/pull/2468#discussion_r565335344 in mind, I wonder if this changes 🤔

Reply:
Seems we should indeed add some kind of  `currentEra` field somewhere.

Maybe we should make this a list then, yes 🤔 

Off the top of my head I would probably just do
```
data ApiEraInfo = ApiEraInfo {
   eraName :: !Era
   start :: !(Maybe ApiEpochInfo)
}
```
since `hardFromEra` doesn't provide any extra information.

Reply:
In-doubt I would stick to the current object format. I think it still has its qualities too. If you feel strongly about it, feel free to disagree.

Reply:
you got me wrong Johannes. I like your proposal, just was thinking loudly and sharing some ideas ;-) no pun intended.great job!